model/qti/metadata/ontology/OntologyMetadataInjector.php

```diff
@@ -26,53 +25,67 @@
 use oat\tao\model\event\MetadataModified;
 use oat\taoQtiItem\model\qti\metadata\MetadataInjector;
 use oat\taoQtiItem\model\qti\metadata\MetadataInjectionException;
-use \core_kernel_classes_Resource;
-use \core_kernel_classes_Property;
-use \InvalidArgumentException;
+use common_Logger;
+use core_kernel_classes_Property;
+use core_kernel_classes_Resource;
+use InvalidArgumentException;
+use Psr\Log\LoggerInterface;
 
 class OntologyMetadataInjector implements MetadataInjector
 {
     private $injectionRules;
-    
-    public function __construct()
-    {
+
+    /** @var EventManager */
+    private $eventManager;
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(
+        LoggerInterface $logger = null,
+        EventManager $eventManager = null
+    ) {
+        $this->eventManager = $eventManager;
+        $this->logger = ($logger ?? common_Logger::singleton()->getLogger());
```

Comment:
No way to avoid static call?

Reply:
No AFAIK :disappointed:  see https://github.com/oat-sa/extension-tao-itemqti/pull/2155/files#r899996806